example/broker.cpp

```diff
@@ -16,18 +16,44 @@
 #include <fstream>
 
 #if defined(MQTT_USE_TLS)
-boost::asio::ssl::context init_ctx(boost::program_options::variables_map const& vm)
+constexpr std::size_t certificate_reload_interval_seconds = 3600;
+
+template<typename ServerPtr>
+void reload_ctx(boost::asio::io_context &ioc, ServerPtr server, boost::program_options::variables_map const& vm, std::string const &name)
 {
+    MQTT_LOG("mqtt_broker", info) << "Reloading certificates for server " << name;
+
     if (vm.count("certificate") == 0 && vm.count("private_key") == 0) {
         throw std::runtime_error("TLS requested but certificate and/or private_key not specified");
     }
 
+    boost::system::error_code ec;
+    server->get_ssl_context().use_certificate_file(vm["certificate"].as<std::string>(), boost::asio::ssl::context::pem, ec);
+    if (ec) {
+        throw std::runtime_error("Failed to load certificate file: " + ec.message());
+    }
+
+    server->get_ssl_context().use_private_key_file(vm["private_key"].as<std::string>(), boost::asio::ssl::context::pem, ec);
+    if (ec) {
+        throw std::runtime_error("Failed to load private key file: " + ec.message());
+    }
+
+    auto reload_timer = std::make_shared<boost::asio::deadline_timer>(ioc, boost::posix_time::seconds(certificate_reload_interval_seconds));
+    reload_timer->async_wait([&, server = server, reload_timer = reload_timer, name = name](boost::system::error_code const &e) {
```

Comment:
Don't use single `&` with async operation.
See https://github.com/redboltz/mqtt_cpp/wiki/Coding-Rules#details-of-lambda-expression

Reply:
Reload timer shouldn't be allocated in the reload_ctx. It should be allocated at run_broker() as the shared_ptr.  And passed by reference to reload_ctx. At the lambda capture, use weak_ptr pattern.

Reply:
`name` shouldn't be copy capture. Use reference capture.

Reply:
`&e` is coding rule violation.

Reply:
> `name` shouldn't be copy capture. Use reference capture.

If you use https://github.com/redboltz/mqtt_cpp/pull/781/files#r552543783 std::string, then name should be move capture.
If char const* then name should be copy capture. copy capture should be `[name]` not `[name = name]`.